akd/src/append_only_zks.rs

```diff
@@ -40,6 +40,33 @@ async fn tic_toc<T>(f: impl core::future::Future<Output = T>) -> (T, Option<f64>
     (f.await, None)
 }
 
+/// An azks is built both by the [crate::directory::Directory] and the auditor.
+/// However, both constructions have very minor differences, and the insert
+/// mode enum is used to differentiate between the two.
+#[derive(Debug, Clone, Copy)]
+pub(crate) enum InsertMode {
+    /// The default mode of constructing the tree
+    Directory,
+    /// The auditor's mode of constructing the tree - last modified epochs of
+    /// leaves are not included in node hashes
+    Auditor,
+}
+
+impl Default for InsertMode {
+    fn default() -> Self {
+        InsertMode::Directory
+    }
+}
```

Comment:
We can actually derive Default as well. In addition adding `Default` on the enum itself, we also need to specify which enum value is the default by annotating it with `#[default]`. See here for reference: https://doc.rust-lang.org/std/default/trait.Default.html#enums.

Reply:
Actually I think we can just get rid of the `impl Default` entirely, and just refer to `InsertMode::Directory()` since it's a purely internal enum and it would be great to be explicit each time it is used.